app/models/incoming_message.rb

```diff
@@ -625,14 +621,7 @@ def extract_attachments
       attachments += uudecode_attachments
     end
 
-    attachment_ids = attachments.map { |attachment| attachment.id }
-    # now get rid of any attachments we no longer have
-    FoiAttachment.
-      where(
-        ["id NOT IN (?) AND incoming_message_id = ?",
-         attachment_ids,
-         self.id]
-      ).destroy_all
+    (foi_attachments - attachments).each(&:mark_for_destruction)
```

Comment:
Given this method is still rather complicated, feel like it warrants preservation of the comment (with some minor rewording)? Something like:

```diff
- # now get rid of any attachments we no longer have
+ # Purge old attachments that have been rebuilt with a new hexdigest
```